src/include/hardware.h

```diff
@@ -25,6 +25,14 @@
         WT_READ_BARRIER();      \
     } while (0)
 
+/*
+ * Read a shared location and guarantee that subsequent reads do not see any earlier state.
+ */
+#define WT_ORDERED_READ64_LIGHTWEIGHT(ret, ptr) \
```

Comment:
The lack of typing associated with this implementation bothers me. Could you take a look at how the `WT_ATOMIC_FUNC` macro is implemented and used, and see whether we could use the same pattern here?

I know there are only 64 bit uses at the moment, but an advantage of that model is that calls to the __wt_atomic_XXX implementations get type checking at the moment.